### PR TITLE
[MM-21901] : Add Interactive Dialog Trigger Timeout config option in Integration Management setting

### LINKED ISSUE #21901: Interactive Dialogs timeout forbiddingly short in slash command integrations
#### Summary
The interactive dialog timeout is hard coded to 3s. 

Integrating with external vendor solutions even basic network as well as application layer latency render integrations nonviable because the dialog will expire before the external system would even have a chance to answer. 

This issue was raised by multiple individuals over the years : https://forum.mattermost.com/t/trigger-id-lifetime/8562/4

While not strictly speaking a bug this limitation severely limits the platform being viable for integrations, which seems highly problematic as vendors such as Slack shine in ease of support for integrations.  Obviously we want MM to be viable for adoption.

#### Steps to reproduce
Build a slash command contacting a slightly slow API. 

#### Expected behavior
Timeout should have a more reasonable default or be entirely configurable. 

#### Observed behavior (that appears unintentional) 
Timeouts

#### Possible fixes
Change constant or expose as ENV variable 

https://github.com/mattermost/mattermost-server/blob/1fe750ac9b1aac6b57de5281252d73e204ce18a4/model/integration_action.go

### PR DESCRIPTION
**Pull request by Apahadi73**
#### Summary

Implement a config option to set the `InteractiveDialogTriggerTimeout` value to allow the client/user to decide the dialogue timeout issue instead of using a constant timeout value

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/21901

#### Related Pull Requests

- Has server changes ([[MM-21901] Set trigger timeout for Interactive Dialogue dynamically using config instead of const value](https://github.com/mattermost/mattermost-server/pull/22485))

#### Release Note

```release-note
Add a config option in the `Integration Management` tab inside `System Console` to allow the client/user to set the `InteractiveDialogTriggerTimeout` value

```

[MM-21901]: https://mattermost.atlassian.net/browse/MM-21901?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: i18n/en.json [hunk 1]
```diff
@@ -2146,6 +2146,9 @@
   "admin.service.googleTitle": "Google API Key:",
   "admin.service.iconDescription": "When true, webhooks, slash commands and other integrations will be allowed to change the profile picture they post with. Note: Combined with allowing integrations to override usernames, users may be able to perform phishing attacks by attempting to impersonate other users.",
   "admin.service.iconTitle": "Enable integrations to override profile picture icons:",
+  "admin.service.InteractiveDialogTriggerTimeout.desc": "The number of milliseconds to wait for a response from the server when triggering an interactive dialog. If the server does not respond within the timeout, the dialog will not be displayed.",
```
Comment (neallred) [inline i18n/en.json:2149]: Nit: Let's keep the casing for these ids consistent with others in the admin definition, i.e. `interactiveDialogTriggerTimeout`

**Review by neallred — CHANGES_REQUESTED** [review]
Functionality looks good to me. Looks like the linter has a few warnings to address: https://github.com/mattermost/mattermost-webapp/actions/runs/4394220896/jobs/7716426062 . I'd also suggest making the casings for the i18n ids consistent with the other ones, and after that, running `make i18n-extract` so that the added strings get sorted correctly.

**Review by mickmister — APPROVED** [review]
Thanks for this improvement @Apahadi73!

LGTM assuming requests above are addressed :+1:

File: i18n/en.json (hunk 1 above)
Comment (Apahadi73) [reply i18n/en.json]: done👍

**Comment by Apahadi73** [comment]
> Functionality looks good to me. Looks like the linter has a few warnings to address: https://github.com/mattermost/mattermost-webapp/actions/runs/4394220896/jobs/7716426062 . I'd also suggest making the casings for the i18n ids consistent with the other ones, and after that, running `make i18n-extract` so that the added strings get sorted correctly.

Hey @neallred, I have implemented your suggestion with the latest PR updates

**Comment by neallred** [comment]
The i18n scripts and automation can be a bit quirky. It looks like when you renamed, something in the order of how things were done made it so the renamed strings had their value deleted: https://github.com/mattermost/mattermost-webapp/pull/12356/files#diff-1a490f08504158838befaa816ee8d5c08dff9e694d5a568ce4b0a4398dcc563eR2152-R2153 . Could you re-fill in those strings to the values they are supposed to be?

**Comment by Apahadi73** [comment]
> The i18n scripts and automation can be a bit quirky. It looks like when you renamed, something in the order of how things were done made it so the renamed strings had their value deleted: https://github.com/mattermost/mattermost-webapp/pull/12356/files#diff-1a490f08504158838befaa816ee8d5c08dff9e694d5a568ce4b0a4398dcc563eR2152-R2153 . Could you re-fill in those strings to the values they are supposed to be?

re-added the string values

**Comment by M-ZubairAhmed** [comment]
Hello there! Could you please open the PR in the mattermost-server as part of the monorepo? I will be closing this PR for now. Thank you very much for your understanding and cooperation in this matter.

**Pull request closed without merging**